src/plugins/saved_objects/public/save_modal/saved_object_save_modal.tsx

```diff
@@ -39,7 +39,11 @@ import { FormattedMessage } from '@kbn/i18n/react';
 import React from 'react';
 import { EuiText } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
-import { VISUALIZE_EMBEDDABLE_TYPE } from '../../../../legacy/core_plugins/visualizations/public/embeddable/constants';
+
+// TODO: can't import from '../../../../legacy/core_plugins/visualizations/public/' directly,
+// because yarn build:types fails after trying to emit type declarations for whole visualizations plugin
+// Bunch of errors like this: 'Return type of exported function has or is using private name 'SavedVis''
+import { VISUALIZE_EMBEDDABLE_TYPE } from '../../../../legacy/core_plugins/visualizations/public/np_ready/public/embeddable/constants';
```

Comment:
// brain pick 
What exactly happened here?

Reply:
As I understood, there is `yarn build:types` which emits d.ts files for `core` code. 
In compilation included only:
```
"include": [
    "src/core/server/index.ts",
    "src/core/public/index.ts",
    "typings"
  ]
```

This file here - `src/plugins/saved_objects/public/save_modal/saved_object_save_modal.tsx` ends up as a deep dependency of `"src/core/public/index.ts"`.

If to import whole `visualisation` plugin, the type declaration emitting breaks, because of incompatible for declarations code as a dependency of `visualisation` plugin. 
The code which causing this error: 
`'Return type of exported function has or is using private name 'SavedVis''`
is code like this:
```
export function createSavedVisLoader(services: SavedObjectKibanaServicesWithVisualizations) {
  const { savedObjectsClient, visualizationTypes } = services;

  class SavedObjectLoaderVisualize extends SavedObjectLoader {
....
 }
return SavedObjectLoaderVisualize
```

See that `SavedObjectLoaderVisualize` is private, so it can't be emitted in declaration. 

This root cause issue should be addressed by: https://github.com/elastic/kibana/issues/58243

P.S. spend like half of a day figuring out what is going wrong :D

Reply:
:tophat: 
:smile: :ok_hand:

Reply:
Wow, thanks for digging into it @Dosant ! This import should definitely not be there, this was an oversight during migration. Saved object save modal shouldn't know anything about the visualize embeddable, this logic should be moved into the specific calls (e.g. by passing in a prop like `hideDescription`). 

I created an issue for this here: https://github.com/elastic/kibana/issues/58479

@joshdover I was under the impression the linter would prevent us from referencing legacy from within the new platform. A quick code search reveals a number of offenders, did something break here?